core/crypto/crypto.c

```diff
@@ -130,25 +133,60 @@ TEE_Result crypto_mac_final(void *ctx __unused, uint32_t algo __unused,
 }
 #endif /*_CFG_CRYPTO_WITH_MAC*/
 
-TEE_Result crypto_authenc_get_ctx_size(uint32_t algo __maybe_unused,
-				       size_t *size __maybe_unused)
+TEE_Result crypto_authenc_alloc_ctx(void **ctx, uint32_t algo)
 {
 	switch (algo) {
 #if defined(CFG_CRYPTO_CCM)
 	case TEE_ALG_AES_CCM:
-		*size = crypto_aes_ccm_get_ctx_size();
-		return TEE_SUCCESS;
+		return crypto_aes_ccm_alloc_ctx(ctx);
 #endif
 #if defined(CFG_CRYPTO_GCM)
 	case TEE_ALG_AES_GCM:
-		*size = crypto_aes_gcm_get_ctx_size();
-		return TEE_SUCCESS;
+		return crypto_aes_gcm_alloc_ctx(ctx);
 #endif
 	default:
 		return TEE_ERROR_NOT_IMPLEMENTED;
 	}
 }
 
+void crypto_authenc_free_ctx(void *ctx, uint32_t algo)
+{
+	switch (algo) {
+	case 0:
```

Comment:
No, `0` is never a valid algorithm. I've added this case to allow the function to be used in the same way as `crypto_hash_free_ctx()`.

The way  is used `crypto_hash_free_ctx()` it's quite convenient to be permitted to call it with algo set to `0` in the cleanup part of the calling function.

Reply:
Sorry, I do not quite understand it. I cannot find an example which call the crypto_authenc_free_ctx() with algo 0. It will be implemented in the future?

Reply:
OK, this allowing `algo` being `0` seems to cause some confusion. I'll remove it.